src/social_db/data/set/function_args/with_json_file.rs

```diff
@@ -0,0 +1,62 @@
+use color_eyre::eyre::Context;
+use inquire::CustomType;
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(input_context = super::super::SetContext)]
+#[interactive_clap(output_context = FunctionArgsContext)]
+pub struct FunctionArgs {
+    #[interactive_clap(skip_default_input_arg)]
+    /// Enter the path to the arguments file:
+    path: near_cli_rs::types::path_buf::PathBuf,
+    #[interactive_clap(named_arg)]
+    /// Specify signer account ID
+    sign_as: super::super::sign_as::Signer,
+}
+
+#[derive(Clone)]
+pub struct FunctionArgsContext(super::ArgsContext);
+
+impl FunctionArgsContext {
+    pub fn from_previous_context(
+        previous_context: super::super::SetContext,
+        scope: &<FunctionArgs as interactive_clap::ToInteractiveClapContextScope>::InteractiveClapContextScope,
+    ) -> color_eyre::eyre::Result<Self> {
+        let data = std::fs::read(&scope.path.0)
+            .wrap_err_with(|| format!("Access to data file <{:?}> not found!", scope.path))?;
+        let value: serde_json::Value =
+            serde_json::from_slice(&data).wrap_err("File data is not in JSON format!")?;
```

Comment:
Use `from_reader` to avoid unnecessary reading to string: https://docs.rs/serde_json/latest/serde_json/fn.from_reader.html (see the example of Open + BufReader + from_reader)

Reply:
Done